binnacle/lib/screens/map_screen.dart

```diff
@@ -0,0 +1,148 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:sos/providers/app_provider.dart';
+
+const int MAX_MARKERS = 16;
+
+class MapScreen extends StatefulWidget {
+  @override
+  _MapState createState() => _MapState();
+}
+
+class _MapState extends State<MapScreen> {
+  Map<PolylineId, Polyline> _polylines = <PolylineId, Polyline>{};
+  Completer<GoogleMapController> _controller = Completer();
+  final Set<Marker> _markers = {};
+  bool _isPlacingPoints = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final bloc = Provider.of(context);
+
+    var _context;
+
+    return MaterialApp(
+        home: Scaffold(
+      body: StreamBuilder(
+          stream: bloc.position,
+          builder: (context, snapshot) {
+            _context = context;
+            if (!snapshot.hasData) {
+              return Center(
+                child: new CircularProgressIndicator(),
+              );
+            } else {
+              _initStartMarker(snapshot.data.latlng);
```

Comment:
Does this constantly keep resetting the start?

Reply:
Fixed! Nice catch :)

This would add a new marker at the start location every time streambuilder did an update.